build.gradle.kts

```diff
@@ -196,3 +197,140 @@ tasks.withType<NodeJsExec>().all {
         )
     }
 }
+
+// WasmEdge tasks
+val wasmEdgeVersion = "0.14.0"
+
+val wasmEdgeInnerSuffix = when (currentOsType.name) {
+    OsName.LINUX -> "Linux"
+    OsName.MAC -> "Darwin"
+    OsName.WINDOWS -> "Windows"
+    else -> error("unsupported os type $currentOsType")
+}
+
+val unzipWasmEdge = run {
+    val wasmEdgeDirectory = "https://github.com/WasmEdge/WasmEdge/releases/download/$wasmEdgeVersion"
+    val wasmEdgeSuffix = when (currentOsType) {
+        OsType(OsName.LINUX, OsArch.X86_64) -> "manylinux_2_28_x86_64.tar.gz"
+        OsType(OsName.MAC, OsArch.X86_64) -> "darwin_x86_64.tar.gz"
+        OsType(OsName.MAC, OsArch.ARM64) -> "darwin_arm64.tar.gz"
+        OsType(OsName.WINDOWS, OsArch.X86_32),
+        OsType(OsName.WINDOWS, OsArch.X86_64) -> "windows.zip"
+        else -> error("unsupported os type $currentOsType")
+    }
+
+    val artifactName = "WasmEdge-$wasmEdgeVersion-$wasmEdgeSuffix"
+    val wasmEdgeLocation = "$wasmEdgeDirectory/$artifactName"
+
+    val downloadedTools = File(buildDir, "tools")
+
+    val downloadWasmEdge = tasks.register("wasmEdgeDownload", Download::class) {
+        src(wasmEdgeLocation)
+        dest(File(downloadedTools, artifactName))
+        overwrite(false)
+    }
+
+    tasks.register("wasmEdgeUnzip", Copy::class) {
+        dependsOn(downloadWasmEdge)
+
+        val archive = downloadWasmEdge.get().dest
```

Comment:
It is better to not use `get()` during configuration phase, only in task actions (`doFirst`, `doLast`).
You can transform providers `downloadWasmEdge.map { it.dest }`
Further in `from(if...)` you can use condition on OS type, not extension (like we do fro Node.js)

Reply:
But because it works now for deno, we can keep this variant